sov-modules/sov-state/src/value.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{storage::StorageKey, Prefix, StateMap, Storage};
+use sovereign_sdk::serial::{Decode, Encode};
+use std::marker::PhantomData;
+
+/// SingletonKey is very similar to the unit type `()` i.e. it has only one value.
+/// We provide a custom efficient Encode implementation for SingletonKey while Encode for `()`
+/// is likely already implemented by an external library (like borsh), which is outside of our control.
+#[derive(Debug)]
+pub struct SingletonKey;
+
+impl Encode for SingletonKey {
+    fn encode(&self, _: &mut impl std::io::Write) {
+        // Do nothing.
+    }
+}
+
+/// Container for a single value.
+#[derive(Debug)]
+pub struct StateValue<V, S> {
+    _phantom: PhantomData<V>,
+    // StateValue is equivalent to a StateMap with a single key.
+    map: StateMap<SingletonKey, V, S>,
+}
```

Comment:
It's a little unintuitive that `StateValue` is a wrapper around `StateMap`. I'd consider either keeping them completely separate, or defining a third type which contains the shared logic and having them both contain that type.

Reply:
I carted a new type `Backend` as suggested.